src/Corax/IndexWriter.cs

```diff
@@ -96,6 +96,14 @@ public void Addition(long entryId)
                 if (FreeSpace == 0)
                     GrowBuffer();
 
+                //Lets assert if it is in last removals
+                if (_removals > 0 && *(_end - _removals + 1) == entryId)
```

Comment:
I don't understand what this code does.

Reply:
Let's have Corax entry like this:
`['item': ["test", "matt"]]` for `doc1`
So, we will create two fields as normal. Then we append something to a list or remove etc.
`['item': ["abc", "matt"]]` for `doc1`.

That means we've two different Corax objects so we will delete the origin terms and append the new us.
So for `matt` the modified entry will look like this:
```
Additions: `doc1`
Removals `doc1`
```
a
Because it happens sequentially we can do a trick and check if the last item is in inverse operation. If it is, we can just delete it and not proceed because there is no real change in the trees.

Reply:
@ayende does it look valid to you?